### PR TITLE
Ridzikowski/travels

### PR DESCRIPTION
**Pull request by ridzikowski**
rename branch and convert OutsideTranssmitionEvent to outsidefeed!() function

### CODE REVIEW

File: src/simparams.jl [hunk 1]
```diff
@@ -42,10 +44,14 @@ struct SimParams
   phone_tracing_params::Union{Nothing, PhoneTracingParams}
 
   infection_modulation_function::Union{Nothing,InfectionModulation}
+  infection_travels_function::Union{Nothing,InfectionTravels}
+  travels_frequency::TimePoint
```
Comment (tozanski) [inline src/simparams.jl:48]: this probably should be a part of the travel function (different functions should be allowed do have different parameters)

File: src/simparams.jl [hunk 2]
```diff
@@ -14,6 +14,8 @@ include("params/strains.jl")
 
 abstract type InfectionModulation end
 
+abstract type InfectionTravels end
```
Comment (tozanski) [inline src/simparams.jl:17]: naming: to me travel is one possible execution of imported cases / external cases; therefore the abstract name should refer to that

File: src/params/travels.jl [hunk 3]
```diff
@@ -0,0 +1,19 @@
...
+    days::Float64 = 0
+    peak::Float64 = 0
+    height::Float64 = 0
+    minimum::Float64 = 0
+  end
+  
+  function (f::Holiday)(state::SimState, params::SimParams, infection_time::Real=0.0)
+    #@assert kind(event) == OutsideTransmissionEvent
+    days = (f.days/2)^2
+    rand(state.rng) < max((Float64(-abs(infection_time-f.peak)^2)*f.height)/days + f.height, f.minimum)
```
Comment (tozanski) [inline src/params/travels.jl:11]: please explain what is this function and perhaps name the function appropriately? if this is a parabola, maybe `ParabolicImports` would be a good name?

File: src/params/travels.jl [hunk 4]
```diff
@@ -0,0 +1,19 @@
...
+  end
+  
+  # This all to avoid using @eval and others
+  const travels = Dict{String, Type{T} where T}(
+      "Holiday" => Holiday
```
Comment (tozanski) [inline src/params/travels.jl:16]: Please add another function to behave similarly as the current `outsidefeed!` to be able to also use that

File: src/MocosSim.jl [hunk 5]
```diff
@@ -94,4 +94,19 @@ function simulate!(state::SimState, params::SimParams, callback)
...
+
+  for infection_time in 0.0:params.travels_frequency:time_limit
+    if params.infection_travels_function(state, params, infection_time)::Bool 
+      person_id = sample(state.rng, individuals)
+      event = Event(Val(OutsideInfectionEvent), infection_time, person_id, strain)
+      push!(state.queue, event)
+    end
+  end
+
+end
```
Comment (tozanski) [inline src/MocosSim.jl:109]: why was it moved here from `simstate.jl`?

**Review by tozanski — CHANGES_REQUESTED** [review]
Very good overall! 

The current changes I would like to still add are mostly naming and parametrization. Other than that please:
- add the a function that corresponds to current `outsidefeed!`
- rebase on current `master` (`git pull origin master && git rebase origin/master`), perhaps a merge from master will also be OK since will squash all the commits anyway
- increment the version number in `Project.toml`

File: src/params/travels.jl (hunk 3 above)
Comment (ridzikowski) [reply src/params/travels.jl:11]: yes, its good name

File: src/simparams.jl (hunk 1 above)
Comment (ridzikowski) [reply src/simparams.jl:48]: function progression is independent of this parameter. function outsidefeed!  push the events in the queue depending on the frequency

File: src/MocosSim.jl (hunk 5 above)
Comment (ridzikowski) [reply src/MocosSim.jl:109]: current version of the function requires that both `simstate` and `simparams` be defined in order to work.

**Review by tozanski — COMMENTED** [review]
Looks good and simple already!

I believe we can still improve the design a little bit to make it even simpler and more generic: Instead of having `outsidefieed!` function that calls the import function inside, let's just merge them and make the import function responsible for everything, including the looping and sampling. 

That would lead to having two implementations of the `imported_cases` function
- one for imports that imports over some time span
- one that imports all the cases in single timepoint

The config could then accept a list of such function so we can have both initial imports that multiple travels happening over various time spans.

Sorry for delaying, but It usually takes few iterations to see all the simplifications and generalizations that are available

**Review by tozanski — APPROVED** [review]
Looks good to me

**Pull request merged**